cmd/kubeadm/app/cmd/upgrade/node.go

```diff
@@ -19,221 +19,253 @@ package upgrade
-	if err := printFilesIfDryRunning(flags.dryRun, kubeletDir); err != nil {
-		return errors.Wrap(err, "error printing files on dryrun")
-	}
+	return &nodeData{
+		etcdUpgrade:        options.etcdUpgrade,
+		renewCerts:         options.renewCerts,
+		dryRun:             options.dryRun,
+		kubeletVersion:     options.kubeletVersion,
+		cfg:                cfg,
+		client:             client,
+		isControlPlaneNode: isControlPlaneNode,
+	}, nil
+}
 
-	fmt.Println("[upgrade] The configuration for this node was successfully updated!")
-	fmt.Println("[upgrade] Now you should go ahead and upgrade the kubelet package using your package manager.")
-	return nil
+// DryRun returns the dryRun flag.
+func (d *nodeData) DryRun() bool {
+	return d.dryRun
 }
 
-// printFilesIfDryRunning prints the Static Pod manifests to stdout and informs about the temporary directory to go and lookup
-func printFilesIfDryRunning(dryRun bool, kubeletDir string) error {
-	if !dryRun {
-		return nil
-	}
+// EtcdUpgrade returns the etcdUpgrade flag.
+func (d *nodeData) EtcdUpgrade() bool {
+	return d.etcdUpgrade
+}
 
-	// Print the contents of the upgraded file and pretend like they were in kubeadmconstants.KubeletRunDirectory
-	fileToPrint := dryrunutil.FileToPrint{
-		RealPath:  filepath.Join(kubeletDir, constants.KubeletConfigurationFileName),
-		PrintPath: filepath.Join(constants.KubeletRunDirectory, constants.KubeletConfigurationFileName),
-	}
-	return dryrunutil.PrintDryRunFiles([]dryrunutil.FileToPrint{fileToPrint}, os.Stdout)
+// RenewCerts returns the dryRun flag.
+func (d *nodeData) RenewCerts() bool {
+	return d.renewCerts
 }
 
-// RunUpgradeControlPlane is executed when `kubeadm upgrade node controlplane` runs.
-func RunUpgradeControlPlane(flags *controlplaneUpgradeFlags) error {
+// KubeletVersion returns the kubeletVersion flag.
+func (d *nodeData) KubeletVersion() string {
+	return d.kubeletVersion
+}
 
-	client, err := getClient(flags.kubeConfigPath, flags.dryRun)
-	if err != nil {
-		return errors.Wrapf(err, "couldn't create a Kubernetes client from file %q", flags.kubeConfigPath)
-	}
+// Cfg returns initConfiguration.
+func (d *nodeData) Cfg() *kubeadmapi.InitConfiguration {
+	return d.cfg
+}
 
-	waiter := apiclient.NewKubeWaiter(client, upgrade.UpgradeManifestTimeout, os.Stdout)
+// IsControlPlaneNode returns the isControlPlaneNode flag.
+func (d *nodeData) IsControlPlaneNode() bool {
+	return d.isControlPlaneNode
+}
 
-	// Fetches the cluster configuration
-	cfg, err := configutil.FetchInitConfigurationFromCluster(client, os.Stdout, "upgrade", false)
-	if err != nil {
-		return errors.Wrap(err, "unable to fetch the kubeadm-config ConfigMap")
-	}
+// Client returns a Kubernetes client to be used by kubeadm.
+func (d *nodeData) Client() clientset.Interface {
+	return d.client
+}
+
+// NewCmdUpgradeNodeConfig returns the cobra.Command for downloading the new/upgrading the kubelet configuration from the kubelet-config-1.X
+// ConfigMap in the cluster
```

Comment:
add TODO to remove when 1.18 is released?

Reply:
I will add an issue as well